sdk/core/src/status_code.rs

```diff
@@ -0,0 +1,59 @@
+use crate::error::{Error, ErrorKind};
+use std::ops::Deref;
+
+#[derive(PartialEq, Clone, Debug)]
+pub struct StatusCode(pub(crate) http_types::StatusCode);
+
+impl Deref for StatusCode {
+    type Target = http_types::StatusCode;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl StatusCode {
+    pub fn as_u16(&self) -> u16 {
+        self.0 as u16
+    }
+    pub fn from_u16(value: u16) -> crate::Result<Self> {
+        Ok(StatusCode(
+            http_types::StatusCode::try_from(value).map_err(|_error| {
+                Error::with_message(ErrorKind::DataConversion, || {
+                    format!("unable to convert {value} to StatusCode")
+                })
+            })?,
+        ))
+    }
+
+    pub const BAD_GATEWAY: StatusCode = StatusCode(http_types::StatusCode::BadGateway);
+    pub const GATEWAY_TIMEOUT: StatusCode = StatusCode(http_types::StatusCode::GatewayTimeout);
+    pub const INTERNAL_SERVER_ERROR: StatusCode =
+        StatusCode(http_types::StatusCode::InternalServerError);
+    pub const NOT_MODIFIED: StatusCode = StatusCode(http_types::StatusCode::NotModified);
+    pub const OK: StatusCode = StatusCode(http_types::StatusCode::Ok);
+    pub const REQUEST_TIMEOUT: StatusCode = StatusCode(http_types::StatusCode::RequestTimeout);
+    pub const SERVICE_UNAVAILABLE: StatusCode =
+        StatusCode(http_types::StatusCode::ServiceUnavailable);
+    pub const TOO_MANY_REQUESTS: StatusCode = StatusCode(http_types::StatusCode::TooManyRequests);
+}
```

Comment:
To be honest, I don't like this construct much. In my opinion status codes are best represented in Rust as a non-exhaustive enum.

I feel like we'd be better off either re-exporting `http_types::Method` as-is, or if we don't want to rely on using `http-types` in our public API, either copying over the implementation - or creating a wrapper/type-alias to it.